### ISSUE TITLE
Support static generation of stubs

### ISSUE THREAD

**Issue by DemiMarie** [jnr/jnr-ffi#100]
It would be nice to support generating stubs statically, at compile-time.  This would improve performance on Android, which doesn't support dynamic class loading.

**Comment by headius** [comment]
I agree, that would be a great feature. But there's nothing in place for that now. Perhaps you can help?

I would suggest we try the following first:

* Generate these stubs with a name that reflects what we're actually binding, parameters etc, so it can be loaded from classloader.
* Attempt to load that name first for each stub.
* If the named class can't be loaded, generate it.
* If a new JNR property is set, dump the genrerated class to the specified filesystem location.

This would allow you to set a flag, run all tests that would generate stubs, and capture the resulting class files. It's not as nice as an offline generation tool, but it's minimally invasive.

Want to try implementing that?

**Comment by headius** [comment]
FWIW this could also be applied to the ASM stubs generated for the other side of JNI; it could attempt to load those stubs as a specifically-named shared library, conditionally dumping that library to disk if it did not already exist. Then you'd be able to ship the ASM stubs ahead of time rather than trying to generate and bind them at runtime (obviously going to fail on some platforms).
